revolutionpi/analog_pin.go

```diff
@@ -17,101 +23,106 @@ type analogPin struct {
 }
 
-func (b *revolutionPiBoard) WriteAnalog(ctx context.Context, pin string, value int32, extra map[string]interface{}) error {
-	analogPin, err := b.controlChip.GetAnalogPin(pin)
-	if err != nil {
-		return err
-	}
-	b.logger.Debugf("Analog: %#v", analogPin)
-	if !analogPin.isAnalogOutput() {
-		return fmt.Errorf("cannot WriteAnalog, pin %s is not an analog output pin", pin)
-	}
-
-	// check to see if analog output is enabled
-	bufOutputRange := make([]byte, 1)
-	outputRangeAddress := analogPin.inputOffset + 69
-	// use the corresponding analog OutputRange pin to check if the analog output is enabled
-	if analogPin.Address == analogPin.outputOffset+2 {
-		outputRangeAddress = analogPin.inputOffset + 79
-	}
-	n, err := analogPin.ControlChip.fileHandle.ReadAt(bufOutputRange, int64(outputRangeAddress))
-	if err != nil {
-		return err
-	}
-	if n != 1 {
-		return fmt.Errorf("unable to determine if pin %s is configured for analog write", pin)
-	}
-	b.logger.Debugf("outputRange Value: %d", bufOutputRange)
-	// at a later date we can use this address to help validate the requested value is within the range of the pin.
-	// for now all we need to check is if the voltage range is not configured.
-	// this results in analog output not being enabled.
-	if bufOutputRange[0] == 0 {
-		return fmt.Errorf("pin %s is not configured for analog write", pin)
-	}
-
-	buf := new(bytes.Buffer)
-	err = binary.Write(buf, binary.LittleEndian, value)
-	if err != nil {
-		return err
-	}
-	return analogPin.ControlChip.writeValue(int64(analogPin.Address), buf.Bytes())
-}
-
 func (pin *analogPin) Write(ctx context.Context, value int, extra map[string]interface{}) error {
 	pin.ControlChip.logger.Debugf("Analog: %#v", pin)
 	if !pin.isAnalogOutput() {
 		return fmt.Errorf("cannot Write to Analog, pin %s is not an analog output pin", pin.Name)
 	}
 
-	// check to see if analog output is enabled
-	outputRangeAddress := pin.inputOffset + 69
-	// use the corresponding analog OutputRange pin to check if the analog output is enabled
-	if pin.Address == pin.outputOffset+2 {
-		outputRangeAddress = pin.inputOffset + 79
-	}
-	bufOutputRange := make([]byte, 1)
-	n, err := pin.ControlChip.fileHandle.ReadAt(bufOutputRange, int64(outputRangeAddress))
-	if err != nil {
-		return err
-	}
-	if n != 1 {
-		return fmt.Errorf("unable to determine if pin %s is configured for analog write", pin.Name)
-	}
-	pin.ControlChip.logger.Debugf("outputRange Value: %d", bufOutputRange)
-	// at a later date we can use this address to help validate the requested value is within the range of the pin.
-	// for now all we need to check is if the voltage range is not configured.
-	// this results in analog output not being enabled.
-	if bufOutputRange[0] == 0 {
-		return fmt.Errorf("pin %s is not configured for analog write", pin.Name)
+	// validate the requested value is within the range of the pin.
+	// NOTE: we currently assume that the output multiplier, divisor, and offset have not been modified
```

Comment:
during planning/scoping was informed not to worry about this, so leaving it as a comment